### ISSUE TITLE
When forceFitColumns=true, some SlickGrid temporary alterations to DOM outside the SlickGrid container remain after initialization

### ISSUE THREAD

**Issue by jonathanjytang** [6pac/SlickGrid#1084]
### Describe the bug

When `forceFitColumns` is set to true, and SlickGrid's container div is within certain types of DOM trees (see reproduction section) that are located within an ancestor with `display: none`, SlickGrid initialization applies a temporary style `{position: 'absolute', visibility: 'hidden', display: 'block'}` on the grid container div's parent that is _not correctly removed_ after initialization is done. When the DOM tree containing SlickGrid's container div is later made visible, the grid container div's parent and all its contents are not visible. (This is related but different than [Issue #568](https://github.com/6pac/SlickGrid/issues/568), the workaround there isn't sufficient.)

The issue stems from two pairs of the SlickGrid methods `cacheCssForHiddenInit` and `restoreCssFromHiddenInit` being called in a nested manner during SlickGrid initialization when `forceFitColumns=true`: the first `cacheCssForHiddenInit` call is at the beginning of SlickGrid initialization, and the second within `resizeCanvas() -> autosizeColumns()` (this "inner" pair of calls to these methods only occurs when `forceFitColumns=true`, and there are no options to disable them and do the work in the application instead). The issue occurs when the list of hidden parents found in the second `cacheCssForHiddenInit` call is different than the list found in the first `cacheCssForHiddenInit` call, which is caused by the temporary CSS changes applied by the first call.
```
cacheCssForHiddenInit() {
  this._hiddenParents = Utils.parents(this._container, ':hidden') as HTMLElement[];
  this._hiddenParents.forEach(el => {
    const old: Partial<CSSStyleDeclaration> = {};
    Object.keys(this.cssShow).forEach(name => {
      if (this.cssShow) {
        old[name as any] = el.style[name as 'position' | 'visibility' | 'display'];
        el.style[name as any] = this.cssShow[name as 'position' | 'visibility' | 'display'];
      }
    });
    this.oldProps.push(old);
  });
}

restoreCssFromHiddenInit() {
  // finish handle display:none on container or container parents
  // - put values back the way they were
  let i = 0;
  if (this._hiddenParents) {
    this._hiddenParents.forEach(el => {
      const old = this.oldProps[i++];
      Object.keys(this.cssShow).forEach(name => {
        if (this.cssShow) {
          el.style[name as CSSStyleDeclarationWritable] = (old as any)[name];
        }
      });
    });
  }
}
```
There seems to be two issues:
1. The `oldProps` array is always appended to in `cacheCssForHiddenInit`, but `restoreCssFromHiddenInit` always takes values from the head of the array, which becomes an issue for subsequent `autosizeColumns` calls if the hidden parents' CSS ever changed since the first `cacheCssForHiddenInit` call (this part means even with `options.suppressCssChangesOnHiddenInit` and doing its equivalent at the application-level, the temporary style is never removed for some grid parents).
2. The second `cacheCssForHiddenInit` call overwrites the `_hiddenParents` array saved by the first `cacheCssForHiddenInit` call, which becomes an issue in this case when the `_hiddenParents` array is different in the first and second `cacheCssForHiddenInit` calls.

I would be happy to help contribute to a PR if needed, would appreciate any suggestions on a general direction for how to address this.

### Reproduction

MRE at [https://stackblitz.com/edit/github-anos6z-ya84v6](https://stackblitz.com/edit/github-anos6z-ya84v6), modified from the SlickGrid StackBlitz demo with parts irrelevant to the MRE deleted.

The "Example 4 Desired Behavior" tab shows what the desired behavior is, while the "Example 4 Actual Behavior" tab shows what actually occurs.

1. The SlickGrid container div is within a parent div with padding, and that parent div is within a parent with `display: none` (the example4.html file)
2. SlickGrid is initialized with option `forceFitColumns=true`
3. The style `{position: 'absolute', visibility: 'hidden', display: 'block'}` applied by SlickGrid during initialization is not removed on some of SlickGrid's parents (eg the ones with padding, whose offsetWidth/offsetHeight becomes non-zero after the first `cacheCssForHiddenInit` call during grid initialization). This results in the grid's entire container div not being displayed properly when the DOM tree containing the grid is eventually displayed. (The MRE is a much-simplified version of the use case within my application, it's just to illustrate the bug)

### Which Framework are you using?

Other

### Environment Info

```shell
SlickGrid 5.14.0

System:
    OS: Windows 11 10.0.22631
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700
    Memory: 26.26 GB / 47.73 GB
  Binaries:
    Node: 18.12.1 - C:\Program Files\nodejs\node.EXE
    npm: 9.1.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 131.0.6778.70
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/6pac/SlickGrid/blob/master/CODE_OF_CONDUCT.md)
- [X] Read the [Wikis](https://github.com/6pac/SlickGrid/wiki).
- [X] Check that there isn't [already an issue](https://github.com/6pac/SlickGrid/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/6pac/SlickGrid/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
hmm it seems surprising that if SlickGrid changes the parent element style, it should at the very least change it back to its original value once it's done initializing the grid. I didn't quite understand everything but certainly a PR is very welcome to help fix the issue, there are plenty of Cypress E2E tests and so if anything goes wrong then the tests should start failing. It's also nice to see that the new Stackblitz I added is starting to be useful to report bugs and provide a repro :)

**Comment by 6pac** [comment]
I think it should be easy enough to add a flag so that repeat calls to ```cacheCssForHiddenInit``` don't have any effect - it should cache the CSS data only once. That's also a good point about a flag to disable this altogether so that the external code can handle it.

**Comment by ghiscoding** [comment]
closed by PR #1085, I'll push a new version in the next few days

**Comment by ghiscoding** [comment]
@jonathanjytang  new release v5.14.3 published, thanks again for the contribution
